islands/opModal.tsx

```diff
@@ -136,270 +141,281 @@ export default function OpModal(
                     <div>
                       <div class="rounded-t flex justify-between">
                         <div class="z-[20] flex items-center justify-start px-4 w-full h-16 bg-web">
-                          {opType === "CONSUMER"
-                            ? <ConsumerIcon className={"mx-2"} />
-                            : <ProducerIcon className={"mx-2"} />}
+                          {iconDisplay()}
                           <div class="flex flex-col">
                             <h3 class="text-lg text-cloud">
-                              {audience?.operationName}
+                              {displayName}
                             </h3>
-                            <p class="text-xs text-cloud">
-                              {`${clients?.length || 0} attached client${
-                                (clients?.length !== 1) ? "s" : ""
-                              }`}
-                            </p>
+                            {displayType === "operation" && (
+                              <p class="text-xs text-cloud">
+                                {`${clients?.length || 0} attached client${
+                                  (clients?.length !== 1) ? "s" : ""
+                                }`}
+                              </p>
+                            )}
                           </div>
                         </div>
                       </div>
-                      <div class="px-4 py-4 rounded-md mx-2">
-                        <div class="mb-2 flex justify-between items-center pr-2">
-                          <h3 class="text-web font-bold text-sm">
-                            Attached Pipelines
-                          </h3>
-                        </div>
-                        {!serviceMap?.pipes?.length
-                          ? (
-                            <a href={"/pipelines"}>
-                              <button class="text-streamdalPurple border border-purple-600 bg-purple-50 font-semibold rounded-lg w-full flex justify-center text-sm px-2 py-1 text-center inline-flex items-center">
-                                <IconPlus class="w-4 h-4 mr-1" />
-                                Create a new pipeline
-                              </button>
-                            </a>
-                          )
-                          : attachedPipeline
-                          ? (
+                      {displayType === "operation"
```

Comment:
Might make sense to break out these drawer parts into different sub-components for operation versus other types remove some of these ternaries.

Reply:
Sounds good. I'm going to merge this in and handle breaking out the opModal component in a different branch.